lib/rclex/generators/msg_c.ex

```diff
@@ -349,20 +349,11 @@ defmodule Rclex.Generators.MsgC do
 
   defp enif_get_builtin("string", var, mbr, term) do
     """
-    unsigned int #{var}_length;
-    #if (ERL_NIF_MAJOR_VERSION == 2 && ERL_NIF_MINOR_VERSION >= 17) // OTP-26 and later
-    if (!enif_get_string_length(env, #{term}, &#{var}_length, ERL_NIF_LATIN1))
-      return enif_make_badarg(env);
-    #else
-    if (!enif_get_list_length(env, #{term}, &#{var}_length))
-      return enif_make_badarg(env);
-    #endif
-
-    char #{var}[#{var}_length + 1];
-    if (enif_get_string(env, #{term}, #{var}, #{var}_length + 1, ERL_NIF_LATIN1) <= 0)
+    ErlNifBinary #{var}_binary;
+    if (!enif_inspect_binary(env, #{term}, &#{var}_binary))
       return enif_make_badarg(env);
 
-    if (!rosidl_runtime_c__String__assign(&(message_p->#{mbr}), #{var}))
+    if (!rosidl_runtime_c__String__assignn(&(message_p->#{mbr}), (const char *)#{var}_binary.data, #{var}_binary.size))
```

Comment:
May this be a typo to `_assign`??

Reply:
Nope. `rosidl_runtime_c__String__assign` and `rosidl_runtime_c__String__assignn` are diffrent.

ref. https://docs.ros.org/en/rolling/p/rosidl_runtime_c/generated/file_include_rosidl_runtime_c_string_functions.h.html#file-include-rosidl-runtime-c-string-functions-h

Reply:
Woops,,, so sorry I didn't know that;;(

Reply:
It's okay! Never mind.